typescript/agentkit/src/action-providers/vaultsfyi/vaultsfyiActionProvider.ts

```diff
@@ -0,0 +1,425 @@
+    const result = await fetch(`${VAULTS_API_URL}/portfolio/wallet-balances?${params.toString()}`, {
+      method: "GET",
+      headers: {
+        "x-api-key": this.apiKey,
+      },
+    });
+    const balances = (await result.json()) as Balances | ApiError;
+    if ("error" in balances) {
+      return `Failed to fetch wallet balances: ${balances.error}, ${balances.message}`;
+    }
+
+    const entries = Object.entries(balances).map(
+      ([network, balances]: [string, Balances[string]]) => {
+        return [
+          network,
+          balances.map(balance => ({
+            address: balance.address,
+            name: balance.name,
+            symbol: balance.symbol,
+            balance: Number(balance.balance) / 10 ** balance.decimals,
+          })),
+        ];
+      },
+    );
+    return JSON.stringify(Object.fromEntries(entries));
+  }
+
+  /**
+   * Returns the users positions.
+   *
+   * @param wallet - The wallet provider instance for blockchain interactions
+   * @returns A record of the users positions
+   */
+  @CreateAction({
+    name: "positions",
+    description: `
+      This action returns the users positions.
+    `,
+    schema: z.object({}),
+  })
+  async positions(wallet: EvmWalletProvider): Promise<string> {
+    const result = await fetch(`${VAULTS_API_URL}/portfolio/positions/${wallet.getAddress()}`, {
+      method: "GET",
+      headers: {
+        "x-api-key": this.apiKey,
+      },
+    });
+    const positions = (await result.json()) as Positions | ApiError;
+    if ("error" in positions) {
+      return `Failed to fetch positions: ${positions.error}, ${positions.message}`;
+    }
+
+    const entries = Object.entries(positions).map(
+      ([network, positions]: [string, Positions[string]]) => {
+        return [
+          network,
+          positions.map(position => ({
+            name: position.vaultName,
+            vaultAddress: position.vaultAddress,
+            asset: {
+              address: position.asset.assetAddress,
+              name: position.asset.name,
+              symbol: position.asset.symbol,
+            },
+            underlyingTokenBalance: Number(position.balanceNative) / 10 ** position.asset.decimals,
+            lpTokenBalance: Number(position.balanceLp) / 10 ** position.asset.decimals,
+            unclaimedRewards: Number(position.unclaimedUsd) > 0,
+            apy: {
+              base: position.apy.base / 100,
+              rewards: position.apy.rewards / 100,
+              total: position.apy.total / 100,
+            },
+          })),
+        ];
+      },
+    );
+    return JSON.stringify(Object.fromEntries(entries));
+  }
+
+  /**
+   * Checks if this provider supports the given network.
+   *
+   * @param network - The network to check support for
+   * @returns True if the network is supported
+   */
+  supportsNetwork(network: Network): boolean {
+    return network.chainId ? Object.keys(VAULTS_NETWORKS).includes(network.chainId) : false;
```

Comment:
Just in case, I would add a check for EVM as well as chainId support.

i.e. `network.protocolFamily == 'evm' && (/** remaining check */)`